clientconn.go

```diff
@@ -133,10 +146,13 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
 		if closeTransport {
 			t.Close()
 		}
-		newTransport, err := transport.NewClientTransport(cc.dopts.protocol, cc.target, cc.dopts.authOptions)
+		newTransport, err := transport.NewClientTransportTimeout(cc.dopts.protocol, cc.target, cc.dopts.authOptions, cc.dopts.timeout)
```

Comment:
Hi, Wes,

Can you hold on a bit and let me finish an entire pass (because I could
revert my previous comments leading to unnecessary work on your side --
pull request review on github is not pleasant ...)? Thanks!

On Tue, Mar 3, 2015 at 2:15 PM, Wes <notifications@github.com> wrote:

> In clientconn.go
> <https://github.com/grpc/grpc-go/pull/88#discussion_r25732724>:
>
> > @@ -133,10 +146,13 @@ func (cc *ClientConn) resetTransport(closeTransport bool) error {
> >  		if closeTransport {
> >  			t.Close()
> >  		}
> > -		newTransport, err := transport.NewClientTransport(cc.dopts.protocol, cc.target, cc.dopts.authOptions)
> > +		newTransport, err := transport.NewClientTransportTimeout(cc.dopts.protocol, cc.target, cc.dopts.authOptions, cc.dopts.timeout)
>
> I was basically mirroring the Dial/DialTimeout approach taken in the net
> package. If you feel it just adds to bloat, though, I'm happy to just
> update NewClientTransportto take a timeout.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/grpc/grpc-go/pull/88/files#r25732724>.
>